### PR TITLE
Destination databricks: Remove user/pass auth

### PR DESCRIPTION
**Pull request by edgao**
closes https://github.com/airbytehq/airbyte-internal-issues/issues/9146

(note: I don't think there's much urgency on this PR, in spite of the ostensible deadline. Databricks is clearly sending comms to all of their users already, so :shrug: hopefully anyone using airbyte+databricks already knows they should switch to oauth)

Databricks will deprecate user/pass auth on july 10

> Databricks-managed passwords will reach end of life on July 10, 2024

but PATs will continue working

> Note that Personal Access Tokens (PATs) will continue to work for API authentication

AFAICT we don't actually need the user+pass? I.e. a PAT is sufficient. (relevant [docs](https://docs.databricks.com/en/integrations/jdbc/authentication.html#databricks-personal-access-token)) So just... stop using the user+pass stuff

Also move oauth to be the first option, since it's the recommended auth type

(I created a databricks PAT + uploaded it into https://console.cloud.google.com/security/secret-manager/secret/SECRET_DESTINATION_DATABRICKS_PERSONAL_ACCESS_TOKEN_CONFIG/versions?project=dataline-integration-testing)

Closes https://github.com/airbytehq/airbyte-internal-issues/issues/9146

### CODE REVIEW

**Comment by edgao** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/40712?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#40692** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/40692?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#40689** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/40689?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#40712** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/40712?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#40567** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/40567?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#40201** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/40201?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#39008** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39008?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#39334](https://github.com/airbytehq/airbyte/pull/39334) <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39334?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#39141** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39141?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#38893** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38893?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#38886** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38886?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @edgao and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-databricks/src/test-integration/kotlin/io/airbyte/integrations/destination/databricks/typededupe/AbstractDatabricksTypingDedupingTest.kt [hunk 1]
```diff
@@ -26,25 +26,25 @@ import org.junit.jupiter.api.BeforeAll
...
+abstract class AbstractDatabricksTypingDedupingTest(
+    private val jdbcDatabase: JdbcDatabase,
+    private val connectorConfig: DatabricksConnectorConfig,
+) : BaseTypingDedupingTest() {
     override val imageName: String
         get() = "airbyte/destination-databricks:dev"
 
     companion object {
-        private var jdbcDatabase: JdbcDatabase = Mockito.mock()
-        private var connectorConfig: DatabricksConnectorConfig = Mockito.mock()
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-databricks/src/test-integration/kotlin/io/airbyte/integrations/destination/databricks/typededupe/AbstractDatabricksTypingDedupingTest.kt:35]: we need these variables to be instanced per connector config, so push them down into their respective class' companion object

File: airbyte-integrations/connectors/destination-databricks/src/test-integration/kotlin/io/airbyte/integrations/destination/databricks/typededupe/AbstractDatabricksTypingDedupingTest.kt [hunk 2]
```diff
@@ -26,25 +26,25 @@ import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.Timeout
 import org.mockito.Mockito
 
-class DatabricksTypingDedupingTest : BaseTypingDedupingTest() {
+abstract class AbstractDatabricksTypingDedupingTest(
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-databricks/src/test-integration/kotlin/io/airbyte/integrations/destination/databricks/typededupe/AbstractDatabricksTypingDedupingTest.kt:29]: adding separate test classes for the oauth and PAT configs, just to verify that they both work. This is super heavyweight :/ but I don't actually know which parts of our code were relying on the user/pass vs the PAT previously, so this lets us test (mostly) all the code paths

Comment (gisripa) [reply airbyte-integrations/connectors/destination-databricks/src/test-integration/kotlin/io/airbyte/integrations/destination/databricks/typededupe/AbstractDatabricksTypingDedupingTest.kt:25]: Reasonable.  Files API and Jdbc both use PAT when switched to that auth mode, so staging and T+D code paths need to be tested.

**Review by gisripa — APPROVED** [review]

**Pull request merged**